### DISCUSSION TITLE
How get data from custom fields in ListPart?

### DISCUSSION THREAD

**Discussion by Diman96** [OrchardCMS/OrchardCore#10106] [Q&A]
I added text field to `ListPart` 
![image](https://user-images.githubusercontent.com/12182810/129239887-6f7da363-7b0e-420c-ae19-277b32135b18.png)
Added `ListPart` to `NewsPage` `ContentType`  and try to get data from text field on `NewsPage-ListPart.liquid` .  How can i get acces to custom fields?

**Comment by Piedone** [comment]
It should be available with `{{ Model.ContentItem.Content.ListPart.YourFieldName.Text }}` But you can also explore what's in your content item like this: https://docs.orchardcore.net/en/latest/docs/reference/modules/Liquid/#content-property

**Reply by Diman96** [reply]
I try this, it doesn`t work, maybe its bug in Ochard.Core ?

**Reply by Piedone** [reply]
What does `<pre>{{ Model.ContentItem.Content }}</pre>` output for you?

**Reply by Diman96** [reply]
Nothing.

**Reply by Diman96** [reply]
If try `{{ Model }}` its output 
![image](https://user-images.githubusercontent.com/12182810/129582721-947f36f3-64dd-43a8-a0f4-a3c2457c7835.png)

**Reply by Piedone** [reply]
Ah, I see List Part's shape template is a bit of an exception from the usual. So my recommendation changes to `{{ Model.ListPart.ContentItem.Content.ListPart.YourFieldName.Text }}`.

BTW I suggest you check out the source of Orchard to be able to figure out these things, in this case this file: https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Lists/Views/ListPart.cshtml

**Comment by ns8482e** [comment]
It's named shape placed under `Model.Content` https://github.com/OrchardCMS/OrchardCore/blob/main/src/docs/reference/modules/Templates/README.md#razor

To access in Razor use `Model.Content.Named("ListPart-MyFieldName")`

**Comment by Diman96** [comment]
@Piedone @ns8482e  I try use razor and liquid but the suggested options do not work, i copy https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Lists/Views/ListPart.cshtml and overided `ListPart.cshtml` in my theme and I can't acces to to my custom field.  `@Model.ListPart.Content` and etc. are just empty.

**Reply by Diman96** [reply]
@sebastienros  @agriffard  Can you please help me, what I'm doing wrong?

**Reply by ns8482e** [reply]
@Diman96  you can't access it from `ListPart` shape because it's not available there as `ListPart` shape's model is C# ViewModel.

The `ListPart-MyFieldName` shape is placed under `Model.Content` of content item's shape,  and only accessible within content item's template i.e. `Content-MyContentType` 

So either create template `ListPart-MyFieldName.cshtml` OR
 access the shape within  `Content-MyContentType.cshtml` using `Model.Content.Named("ListPart-MyFieldName")`
